### PR TITLE
Allowed max motors to be set in target.h. Set to 6 for CC3D.

### PR DESCRIPTION
**Pull request by martinbudden**
Allowed target specific setting of max motor count. Set to 6 for CC3D - saves about 850 bytes of ROM for CC3D build. CC3D has only 6 PWM pins anyway.

### CODE REVIEW

File: src/main/drivers/pwm_mapping.h [hunk 1]
```diff
@@ -19,18 +19,27 @@
...
 #define MAX_MOTORS  4
 #define MAX_SERVOS  1
+#define MAX_PWM_OUTPUT_PORTS MAX_PWM_MOTORS // must be set to the largest of either MAX_MOTORS or MAX_SERVOS
+
+#elif defined(TARGET_MOTOR_COUNT)
+#define MAX_PWM_MOTORS TARGET_MOTOR_COUNT
+#define MAX_PWM_SERVOS 8
+#define MAX_MOTORS  TARGET_MOTOR_COUNT
+#define MAX_SERVOS  8
+#define MAX_PWM_OUTPUT_PORTS MAX_PWM_SERVOS // must be set to the largest of either MAX_MOTORS or MAX_SERVOS
```
Comment (ledvinap) [inline src/main/drivers/pwm_mapping.h]: Any reason to hardcode this? MAX should work here fine. And  MAX_PWM_OUTPUT_PORTS probably does not need to be public.

**Comment by ledvinap** [comment]
Maybe TARGET_SERVO_COUNT can be added too

**Comment by martinbudden** [comment]
@ledvinap , good observations, thanks.

I looked at also setting servo count, but this is more complicated, since currently servers are allocated specific function (e.g. `SERVO_FLAPS = 7`) so you cannot reduce the number of servos without breaking the airplane mixer. Certainly something to look at in future though.

File: src/main/target/CC3D/target.h [hunk 2]
```diff
@@ -126,9 +124,15 @@
...
+#define BLACKBOX
+#define ENABLE_BLACKBOX_LOGGING_ON_SPIFLASH_BY_DEFAULT
+
+#define TELEMETRY
+#define SERIAL_RX
+#define USE_SERVOS
+#define USE_CLI
+#define TARGET_MOTOR_COUNT 6
+
+
```
Comment (hydra) [inline src/main/target/CC3D/target.h]: just a note to say that re-arranging code as part of a PR makes it harder to review the changes.  If cleanups can be done in different PR's that'd be help for speedier code reviews for functional changes.

**Pull request merged**